sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -653,13 +653,15 @@ private[sql] class RocksDBStateStoreProvider
    * @param uniqueId Optional unique identifier for checkpoint
    * @param readOnly Whether to open the store in read-only mode
    * @param existingStore Optional existing store to reuse instead of creating a new one
+   * @param forceSnapshotOnCommit Whether to force a snapshot upload on commit
```

Comment:
Lets also add a metrics for both hdfs and rocksdb, to show that we forced snapshot. This will help us know the number of stores that snapshot was forced for during the batch. So that it shows up in the streaming query progress metrics. [See](https://github.com/apache/spark/blob/6d5f111cc86a0194130d77510a4d739e32c875df/sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala#L437)